packages/synapse-react-client/src/lib/containers/oauth/CreateOAuthClient.tsx

```diff
@@ -199,246 +214,224 @@ export const CreateOAuthModal: React.FunctionComponent<
-                )}
-              </Row>
-              <Row>
-                <Col lg={6} md={6} sm={12} xs={12}>
-                  <Form.Label>Link to Privacy Policy</Form.Label>
-                  <Form.Control
-                    onChange={e => setPolicyUri(e.target.value)}
-                    placeholder="https://"
-                    type="text"
-                    value={policyUri}
-                  />
-                </Col>
-                <Col lg={6} md={6} sm={12} xs={12}>
-                  <Form.Label>Links to Terms of Service</Form.Label>
-                  <Form.Control
-                    onChange={e => setTosUri(e.target.value)}
-                    placeholder="https://"
-                    type="text"
-                    value={tosUri}
-                  />
-                </Col>
-              </Row>
-              {isEdit && (
-                <div className="danger">
-                  <Typography
-                    style={{ marginTop: '8px' }}
-                    color="error"
-                    variant="headline3"
-                  >
-                    DANGER ZONE
-                  </Typography>
-                  <Typography variant="smallText1">
-                    Editing the following information will render your client
-                    invalid and will require you to create it again and resubmit
-                    verification if needed.
-                  </Typography>
-                  <Row>
-                    <Col lg={6} md={6} sm={12} xs={12}>
-                      <Form.Label htmlFor="redirect-uri-0">
-                        Redirect URI(s)
-                      </Form.Label>
-                      <Tooltip title={uriHelpMessage} placement="top">
-                        <HelpOutlineTwoTone className={`HelpButton`} />
-                      </Tooltip>
+  const redirectURIInputs = (
+    <StyledFormControl fullWidth variant="standard" margin="normal">
+      <InputLabel
+        htmlFor="redirect-uri-0"
+        required
+        sx={{
+          color: isEdit ? 'error.main' : undefined,
+          fontWeight: 700,
+          mb: '4px',
+          pointerEvents: 'auto',
+          zIndex: 1,
+        }}
+      >
+        Redirect URI(s)
+        <Tooltip title={uriHelpMessage} placement="top">
+          <HelpOutlineTwoTone
+            className={`HelpButton`}
+            sx={{ float: 'right' }}
+          />
+        </Tooltip>
+      </InputLabel>
+      {redirectUris?.map((singleUri, idx) => (
+        <div key={idx}>
+          <Box display="flex" gap="2px">
+            <MuiTextField
+              name="uri"
+              fullWidth
+              id={`redirect-uri-${idx}`}
+              onChange={e => handleUriChange(e, idx)}
+              value={singleUri.uri}
+              placeholder="https://"
+              type="text"
+            />
+            {redirectUris.length > 1 && (
+              <IconButton
+                sx={{ mt: '24px', width: '50px' }}
```

Comment:
Add margin to the IconButton so that it is positioned inline with the input, rather than the entire TextField. Also, specify a width so that the delete button has a circle background on hover (vs an ellipse).

Reply:
Can you just do `alignItems: 'center'` on the flex container instead of the custom margin?

Reply:
Potentially! For ease of integration with MUI's TextField, I moved the delete icon to an endAdornment, which eliminates the custom styling. Happy to move the delete icon outside of the field, if that's preferred though!

<img width="412" alt="image" src="https://github.com/Sage-Bionetworks/synapse-web-monorepo/assets/26949006/c7ee623a-8114-4088-a9c1-41da18202beb">

Reply:
Great! Yes, the button still works and the input handles a long URL --

https://github.com/Sage-Bionetworks/synapse-web-monorepo/assets/26949006/60fe0abc-01e2-451d-b107-b936512372e4